drivers/counter/counter_mcux_rtc.c

```diff
@@ -133,20 +133,21 @@ static int mcux_rtc_cancel_alarm(struct device *dev, u8_t chan_id)
 	return 0;
 }
 
-static int mcux_rtc_set_top_value(struct device *dev, u32_t ticks,
-				  counter_top_callback_t callback,
-				  void *user_data)
+static int mcux_rtc_set_top_value(struct device *dev,
+				  const struct counter_top_cfg *cfg)
 {
 	const struct counter_config_info *info = dev->config->config_info;
 	struct mcux_rtc_data *data = dev->driver_data;
 
-	if (ticks != info->max_top_value) {
-		LOG_ERR("Wrap can only be set to 0x%x", info->max_top_value);
+	if ((cfg->ticks != info->max_top_value) ||
+		!(cfg->flags & COUNTER_TOP_CFG_DONT_RESET)) {
```

Comment:
By default `COUNTER_TOP_CFG_DONT_RESET` is not set so this if statement will evaluate to true. This is change of behavior. I didn't check the datasheet but how come this counter doesn't start counting from 0 when it's reaching the top value?

Reply:
i've digged into the manual. Counter can be reset or left free running. I've added implementation to support resetting the counter.

> This is change of behavior

well, behavior was a bug according to `counter_set_top_value` description in master:
`Function sets top value and resets the counter to 0 or top value depending on counter direction.`

Now it will be clarified.